### PR TITLE
Updates docs for Native and VM dev environments

### PR DESCRIPTION
**Pull request by tyler-ball**
Signed-off-by: Blake Stier <bstier@chef.io>
Signed-off-by: Tyler Ball <tball@chef.io>

### CODE REVIEW

**Comment by thesentinels** [comment]
Thanks for the pull request! Here is what will happen next:
 1. Your PR will be reviewed by the maintainers
 2. If everything looks good, one of them will `approve` it, and your PR will be merged.

Thank you for contributing!

File: BUILDING.md [hunk 1]
```diff
@@ -204,7 +205,7 @@ sudo su - jdoe
...
+While the available docker images can provide a convenient contributor onboarding experience, they may not prove ideal for extensive habitat development. Building habitat components is a disk intensive operation. Mounted file systems accross network boundaries, even when confined to a single local host, can yield build times dramatically slower than building against a locally attached file system. Build times will be best on either a bare metal linux environment or a dedicated linux vm. Also note that when using vagrant to provision a vm for habitat development, it is strongly advised that you do not put your habitat repo on a synced folder or at least do not use that folder as your build target. The following directions clone the habitat repository into a local directory to realize faster build times.
```
Comment (cnunciato) [inline BUILDING.md]: Little typo: `accross` - `across` (since you're in there). ;)

**Comment by cnunciato** [comment]
I know some folks do clone into the VM, but I don't personally feel comfortable suggesting this as a default way of working on Habitat, since if the VM aborts or gets into an otherwise unrecoverable state (which happens to me more often than I'd like), you lose all your work. So I'd personally prefer we put these instructions in a place more clearly identifying them as "do this at your own risk".

**Comment by tashimi** [comment]
tenor-26606967

**Comment by tashimi** [comment]
@thesentinels approve

**Comment by thesentinels** [comment]
:metal: I am testing your branch against master before merging it. We do this to ensure that the master branch is never failing tests.

**Comment by thesentinels** [comment]
:neckbeard: Travis CI has [started testing this PR](https://travis-ci.org/habitat-sh/habitat/builds/277238545).

**Comment by thesentinels** [comment]
:sparkling_heart: Travis CI [reports this PR passed](https://travis-ci.org/habitat-sh/habitat/builds/277238545).

It always makes me feel nice when humans approve of one anothers work. I'm merging this PR now.

I just want you and the contributor to answer me one question:

gif-keyboard-3280869874741411265

**Pull request merged**